packages/Python/lldbsuite/test/functionalities/ubsan/user-expression/TestUbsanUserExpression.py

```diff
@@ -0,0 +1,49 @@
+"""
+Test that hitting a UBSan issue while running user expression doesn't break the evaluation.
+"""
+
+import os
+import time
+import lldb
+from lldbsuite.test.lldbtest import *
+from lldbsuite.test.decorators import *
+import lldbsuite.test.lldbutil as lldbutil
+import json
+
+
+class UbsanUserExpressionTestCase(TestBase):
+
+    mydir = TestBase.compute_mydir(__file__)
+
+    @skipUnlessUndefinedBehaviorSanitizer
+    def test(self):
+        self.build()
+        self.ubsan_tests()
+
+    def setUp(self):
+        # Call super's setUp().
+        TestBase.setUp(self)
+        self.line_breakpoint = line_number('main.c', '// breakpoint line')
+
+    def ubsan_tests(self):
+        # Load the test
+        exe = os.path.join(os.getcwd(), "a.out")
+        self.expect(
+            "file " + exe,
+            patterns=["Current executable set to .*a.out"])
+
+        self.runCmd("breakpoint set -f main.c -l %d" % self.line_breakpoint)
+
+        self.runCmd("run")
+
+        process = self.dbg.GetSelectedTarget().process
+        thread = process.GetSelectedThread()
```

Comment:
It is better to set the breakpoint with SBTarget::BreakpointSetByLocation, which returns the breakpoint set, and then assert that you've hit the breakpoint you intended, and that the thread you got is actually the thread that hit the breakpoint, by using lldbutil.get_threads_stopped_at_breakpoint.  That way if something happens like your main thread hits a breakpoint but some secondary thread some system decided ought to always be created crashes, you will see that right away.

I added an lldbutil routine to llvm.org (lldbutil.run_to_source_breakpoint) that packages all this up more conveniently.  When we merge that over to github, that will be the preferred way to gather up all this boilerplate.